larpandoracontent/LArControlFlow/TrackDirectionTool.h

```diff
@@ -0,0 +1,1089 @@
+/**
+ *  @file   larpandoracontent/LArControlFlow/TrackDirectionTool.h
+ *
+ *  @brief  Header file for the track direction finding tool AlgorithmTool class.
+ *
+ *  $Log: $
+ */
+#ifndef LAR_TRACK_DIRECTION_TOOL_H
+#define LAR_TRACK_DIRECTION_TOOL_H 1
+
+#include "larpandoracontent/LArObjects/LArTwoDSlidingFitResult.h"
+#include "larpandoracontent/LArObjects/LArThreeDSlidingFitResult.h"
+#include "larpandoracontent/LArObjects/LArPfoObjects.h"
+#include "larpandoracontent/LArHelpers/LArPfoHelper.h"
+#include "Pandora/AlgorithmTool.h"
+
+#include "larpandoracontent/LArControlFlow/MasterAlgorithm.h"
```

Comment:
I altered the interface as suggested, but it still needs the MasterAlgorithm.h to run, otherwise errors are seen, many of which are "error: ‘class lar_content::TrackDirectionTool’ has no member named ‘GetPandora’".

Reply:
Hi, This is presumably because the MasterAlgorithm.h header file currently defines the `TrackDirectionBaseTool` class, from which the `TrackDirectionTool` inherits. It might be time to factor out the tool definitions into (a) separate header(s) and see where we can get away with forward declarations instead of definitions.